### DISCUSSION TITLE
E-Fil+ Proposal: Designating a "Lead SP"

### DISCUSSION THREAD

**Discussion by kevzak** [filecoin-project/notary-governance#582] [E-Fil+ WG]
For discussion from the details of the main E-Fil+ issue: https://github.com/filecoin-project/notary-governance/issues/559

It is being proposed that for enterprise clients a "Lead SP" will be in charge of:

- Sourcing other SPs that have comparable ops, service levels + infrastructure. 
- Managing service level agreements + contracts. 
    - This includes the service the client requires to support redundancy, distribution, retrieval, data security & sustainability are defined as part of deal making. 
     - This insures the client is bound contractually up front including clarity of consequences of system abuse.
- Organizing and submitting distribution plans:
   - All SP geographies
   - Client data transfer plan e.g. network/ offline
   - Distribution timeframes

As a result, a lead SP is eligible to:
- Earn 10x power for verified deals (as is today)
- And entitled to a “lead origination bonus” where they can store 2x copies (each copy in different locations)

**Comment by kevzak** [comment]
This week, we held discussions about the role and potential reward for a "Lead SP" for enterprise clients. Based on a survey to the Notary WG and SP's who have applied for the E-Fil+ Pilot: 78% said that, yes, SPs who are onboarding an enterprise client could receive an extra copy of a dataset as an organizational bonus for the work.

However, the feedback stated that guidelines should be followed by designated Lead SPs: 
-If there are 5 total copies, the lead SP can have no more than 40% (2 copies) and each additional SP can have no more than 20%
-If the Lead SP can provide more than two nodes in different locations, it is possible to store two backups
-If less copies required by client (ex: only 3) - max is one copy (33%)

In other feedback:
-Several community members voiced that an independent lead reward (ex: FIL) as opposed to multiple copies could be a reward. (Disconnecting rights to extra copies as a reward)
-Others stated that clients should pay a reward for onboarding support.

**Comment by kevzak** [comment]
Additionally, we had a good discussion last week among SPs about Enterprise needs. The feedback was that Enterprise is and will be completely different use case then we've seen before and the decisions about storing copies is ultimately up to the client. 

Yes, if an SP has a client, they will likely lead by default and want to take on the "Lead SP" responsibilities we've identified to ensure their client has a positive experience. However, Enterprise clients want qualified Enterprise SPs, not just to find miners to store their data. "Qualified" includes things like having a Data Center, proper insurance coverages, being certified and meeting Enterprise client standards. 

Therefore, storage of the data and required copies it not about rewarding a "Lead SP" with one extra copy, but will be more so about the client requirements and/or available Enterprise SPs that clients want to use. This should drive how copies are designated and stored for Enterprise.

For example, some of our "Enterprise qualified" SPs have multiple locations and technically if their client wanted two copies in US and one in Canada, they could store decentralized copies in LA, Las Vegas and Toronto (as an example) if they are the only Enterprise SP and that meets the decentralized requirement of storage.

For the pilot we will need to work through details with each SP to define if they will take the lead and to ensure the client requirements can be met by available Enterprise qualified SPs. Copy details will need to be detailed in the application and discussed for approval.

**Reply by MegTei** [reply]
This makes sense @kevzak enterprise deals need to reflect the clients needs/ reqs (so long as established as a trusted party) .

The working assumption does remain, that a Lead SP does a lot more work and will be establishing something like a 'service level agreement' with a client so its reasonable to offer some form of origination bonus/ reward/ incentive/ program (like Gravity) to compensate the Lead SP for the effort bringing verified data to the network.
